posts/a-guinea-pig-in-the-cloud.md

```diff
@@ -0,0 +1,389 @@
+
+Rather than going through a manual construction of this infrastructure thanks to the AWS Web console, I have privilege an automated approach based on [terraform](https://www.terraform.io/) scripts, largely easier to read than their [CloudFormation](https://aws.amazon.com/fr/cloudformation/) counterparts. All necessary files are available in the cloud directory of the gitlab project: [https://gitlab.ippon.fr/bpinel/ember-aws-ehipster/tree/master/cloud](https://gitlab.ippon.fr/bpinel/ember-aws-ehipster/tree/master/cloud) that you should get on your local computer.
+
+Once this is done, provisioning the whole platform is as simple as running the following terraform command:
+
+```
+terraform init
+terraform apply
+```
+
+As already stated before, the terraform scripts create the following resources :
+
+*   3 S3 buckets, one for storing the lambda code (named _lambda-jsonapi-code-bucket_) and the two other for receiving the 'staging' and 'production' version of the application (respectively named _ember-aws-ehipster-staging_ and _ember-aws-ehipster-production_)
+*   1 IAM role named _lambda_jsonapi_ for giving access to the DynamoDB database to the lambda function
+*   1 API gateway named _JsonApiRestGateway_ providing a subset of the JSON API specification
+*   1 lambda function named _lambda-jsonapi_ dealing with the 
+
+Going to your AWS console, on the API Gateway page, you should get a new JsonApiRestGatewayAPI:
+
+![API Gateway web console](https://github.com/bpinel/blog-usa/blob/master/images/2019/02/api-gateway-console-01.png)
+
+Clicking on this API and going to its Staging, you should find a 'staging' deployment. You should grab here its Invoke URL:
+
+![API Gateway web console](https://github.com/bpinel/blog-usa/blob/master/images/2019/02/api-gateway-console-02.png)
+
+As a matter of fact, you need to tell your Ember application to use this URL as the Gateway for all calls done though Ember Data. Everything happens in the file_ config/environment.js_ in the self explanatory lines:
+
+```javascript
+ if (environment === 'production') {
+    // here you can enable a production-specific feature
+    ENV.gatewayURL = 'https://xxxxxxxxxxx.execute-api.us-east-1.amazonaws.com/staging';
+    //ENV.gatewayURL = process.env.STAGING_GATEWAY_URL;
+    console.log("Setting staging gatewayURL to "+ENV.gatewayURL);
+  }
+  if (environment === 'staging') {
+    // here you can enable a staging-specific feature
+    ENV.gatewayURL = 'https://xxxxxxxxxxx.execute-api.us-east-1.amazonaws.com/staging';
+    //ENV.gatewayURL = process.env.PRODUCTION_GATEWAY_URL;
+    console.log("Setting production gatewayURL to "+ENV.gatewayURL);
+  }
+```
+
+### A little test before going further
+
+To check the good provisioning of the cloud infrastructure, you can easily launch a test. To do that, just go to the API Gateway Web console, look for the newly created _JsonApiRestGateway_ API and especially to the GET méthod for /{type} and click on the 'Test' button:
```

Comment:
This is also awkwardly phrased... and has a sneaky French survivor toward the end. 😄 

```suggestion
To verify that the cloud infrastructure is properly provisioned, you can open the API Gateway Web console, look for the newly created `JsonApiRestGateway` API –– specifically, the `GET` method for `/{type}` –– and click on the **Test** button:
```